### PR TITLE
docs: Add architecture diagram to website

### LINKED ISSUE #399: add architecture diagram
**Describe the solution you'd like**
add an architecture diagram

we have a few old ones here: https://docs.google.com/document/d/1Rz1bkZKZSLVMjC_w8WLASPDUjfU80tjV-XWUXZ8vq3I/edit#heading=h.bbpo3iwqk0xy

old draw.io diagrams:
https://drive.google.com/file/d/1sXrFWtdbWPrPSzT94aXpsXTc0fg-dFnF/view?usp=sharing
https://drive.google.com/file/d/1L_wDwU9zKaLuv5qBmoTiYx4NUqForUL-/view?usp=sharing

### PR DESCRIPTION
**Pull request by pmengelbert**
Signed-off-by: Peter Engelbert <pmengelbert@gmail.com>

**What this PR does / why we need it**:
Adds a high-level diagram of Eraser's architecture

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #399

### CODE REVIEW

**Review by ashnamehrotra — APPROVED** [review]
LGTM

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (sozercan) [inline yarn.lock]: we don't need this file

File: docs/docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+---
+title: Architecture
+---
+# Eraser architecture
```
Comment (sozercan) [inline docs/docs/architecture.md]: title is already a top level title here

File: docs/docusaurus.config.js [hunk 3]
```diff
@@ -7,7 +7,7 @@ const darkCodeTheme = require('prism-react-renderer/themes/dracula');
 /** @type {import('@docusaurus/types').Config} */
 const config = {
   title: 'Eraser Docs',
-  url: 'https://azure.github.io/eraser',
+  url: 'https://azure.github.io',
```
Comment (sozercan) [inline docs/docusaurus.config.js]: did you test this change? i think this will break the website gen

**Review by sozercan — CHANGES_REQUESTED** [review]

**Comment by sozercan** [comment]
can we replace Qualys (since it doesn't exist) with `Other providers` or something?

File: docs/docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
```
Comment (sozercan) [inline docs/docs/architecture.md:14]: can we add a brief description here? we can take from google docs probably

let's call out metrics are not available in v0.4

File: docs/docusaurus.config.js (hunk 3 above)
Comment (pmengelbert) [reply docs/docusaurus.config.js]: Was getting a warning about this. I'll change it back

File: docs/docs/architecture.md (hunk 2 above)
Comment (pmengelbert) [reply docs/docs/architecture.md]: What do you mean?

Comment (sozercan) [reply docs/docs/architecture.md]: I mean Architecture will be a top level title on the page, we don't need to add `# Eraser architecture` again

**Review by sozercan — APPROVED** [review]
minor nit, otherwise LGTM

**Pull request merged**